packages/uploadthing/src/sdk/index.ts

```diff
@@ -55,8 +60,40 @@ export const getFileUrls = async (fileKeys: string[] | string) => {
     },
     body: JSON.stringify({ fileKeys }),
   });
+
   if (!res.ok) {
     throw new Error("Failed to get file urls");
   }
+
   return res.json().then(({ data }) => data as { key: string; url: string }[]);
 };
+
+/**
+ * Request file list from UploadThing storage.
+ * @param {string | string[]} fileKeys
+ *
+ * @example
+ * const data = await listFiles();
+ * console.log(data); // {fileKeys: ["2e0fdb64-9957-4262-8e45-f372ba903ac8_image.jpg"], fileIds: ["2e0fdb64-9957-4262-8e45-f372ba903ac8"]}
+ */
+export const listFiles = async () => {
+  if (!UT_SECRET) throw new Error("Missing UPLOADTHING_SECRET env variable.");
+
+  // TODO: Implement filtering and pagination
+  const res = await fetch(generateUploadThingURL("/api/listFiles"), {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      "x-uploadthing-api-key": UT_SECRET,
+      "x-uploadthing-version": UPLOADTHING_VERSION,
+    },
+  });
+
+  if (!res.ok) {
+    throw new Error("Failed to list files");
+  }
+
+  return res
+    .json()
+    .then(({ data }) => data as { fileKeys: string[]; fileIds: string[] });
```

Comment:
Should the return type not be `{ key: string, id: string }[]`

Reply:
Currently that endpoint is actually returning

```
{
  fileIds: {
     fileIds: string[]
     fileKeys: string[]
  }
}
```

🤦 It should probably return something more reasonable like you describe though....

Reply:
![image](https://github.com/pingdotgg/uploadthing/assets/11494384/552a1490-36ba-4275-b065-adfb7e375697)